src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -204,7 +204,7 @@ private async Task StartAsyncCore(TransferFormat transferFormat, CancellationTok
                 return;
             }
 
-            await _connectionLock.WaitAsync();
+            await _connectionLock.WaitAsync(cancellationToken);
```

Comment:
We can remove this if it blocks us from getting this in ask mode. Not totally related to this change but it'c correct.

Reply:
Do we know what the behavior would be if we cancelled waiting for the lock. Does `WaitAsync` throw if it's cancelled?

Reply:
Yes, it throws an `OperationCanceledException`

Reply:
We actually miss this in a number of places. I'll just remove it from here, file an issue and follow up.

Reply:
> Yes, it throws an `OperationCanceledException`

Right, but what happens if I try to `StartAsync` again after the cancellation? Is the object just broken when you cancel `StartAsync`?

Reply:
Good question. I hadn't thought of that. It looks like it will reevaluate any cancellation token you pass into it. It should evaluate call to `WaitAsync` independently.

https://referencesource.microsoft.com/#mscorlib/system/threading/SemaphoreSlim.cs,612

Reply:
> We actually miss this in a number of places. I'll just remove it from here, file an issue and follow up.

Sounds good, let's do this for now.